hpobench/benchmarks/mo/lm_benchmark.py

```diff
@@ -0,0 +1,368 @@
+
+        t = tqdm.tqdm(total=epochs)
+        for epoch in range(epochs):
+            epoch_start_time = time.time()
+            train_loss, train_acc = model.train_fun(self.ntokens, criterion, train_data, learning_rate, clip)
+            train_time += time.time() - epoch_start_time
+            start = time.time()
+            val_loss, val_acc = model.eval_fun(self.ntokens, criterion, val_data)
+            val_loss = np.clip(val_loss, 1e-10, 10)
+            print("val acc for last epoch", val_acc)
+            eval_time += start - time.time()
+
+            t.set_postfix(val_accuracy=val_acc)
+            t.update()
+
+            if not np.isfinite(val_loss):
+                val_loss = 7
+
+            # Save the model if the validation loss is the best we've seen so far.
+            if not best_val_loss or val_loss < best_val_loss:
+                best_val_loss = val_loss
+            else:
+                # Anneal the learning rate if no improvement has been seen in the validation dataset.
+                learning_rate /= learning_rate_factor
+
+        start_time = time.time()
+        _, test_acc = model.eval_fun(self.ntokens, criterion, test_data)
+        eval_test_runtime = time.time() - start_time
+
+        perplexity = math.exp(best_val_loss)
+        log_perplexity = best_val_loss
+        neg_log_perplexity = 10 - best_val_loss
+        elapsed_time = float(ts_start - time.time())
+
+        return {'function_value': {'log_perplexity': log_perplexity,
+                                   'accuracy': val_acc.item(),
+                                   'time': train_time + eval_time
+                                   },
+                'cost': elapsed_time,
+                'info': {'train_accuracy': train_acc.item(),
+                         'validation_accuracy': val_acc.item(),
+                         'test_accuracy': test_acc.item(),
+                         'log_perplexity': log_perplexity,
+                         'perplexity': perplexity,
+                         'negative_log_perplexity': neg_log_perplexity,
+                         'training_cost': train_time,
+                         'valid_cost': eval_time,
+                         'test_cost': eval_test_runtime,
+                         'fidelity': fidelity
+                         }
+                }
+
+    @AbstractMultiObjectiveBenchmark.check_parameters
+    def objective_function_test(self, configuration: Union[CS.Configuration, Dict],
+                                fidelity: Union[Dict, None] = None,
+                                rng: Union[np.random.RandomState, int, None] = None,
+                                shuffle: bool = False,
+                                **kwargs) -> Dict:
+        """
+        Get the validated results. Runs a given configuration on the largest budget (here: 50).
+        Parameters
+        ----------
+        configuration
+        fidelity: Dict, None
+            epoch: int - Values: [1, 81]
```

Comment:
We only allow 81. 

Maybe this is something we should discuss: 
It could be okay to query also the performance for different epoch steps